src/firmware/app/control/trajectory_planner.c

```diff
@@ -168,44 +36,52 @@ app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
-                variable_time_trajectory->trajectory_elements[i - 1].position.y;
-
-            const float interpolated_speed =
-                slope_speed *
-                    (trajectory_time -
-                     variable_time_trajectory->trajectory_elements[i - 1].time) +
-                variable_time_trajectory->speed_profile[i - 1];
+                current_element->position.y - previous_element->position.y;
+            const float delta_theta =
+                current_element->orientation - previous_element->orientation;
+            const float delta_omega =
+                current_element->angular_speed - previous_element->angular_speed;
+            const float delta_linear_speed =
+                current_element->linear_speed - previous_element->linear_speed;
+
+            const float interpolated_x = shared_tbots_math_linearInterpolation(
+                previous_element->time, previous_element->position.x,
+                current_element->time, current_element->position.x, trajectory_time);
+            const float interpolated_y = shared_tbots_math_linearInterpolation(
+                previous_element->time, previous_element->position.y,
+                current_element->time, current_element->position.y, trajectory_time);
+            const float interpolated_theta = shared_tbots_math_linearInterpolation(
+                previous_element->time, previous_element->orientation,
+                current_element->time, current_element->orientation, trajectory_time);
+            const float interpolated_angular_speed =
+                shared_tbots_math_linearInterpolation(
+                    previous_element->time, previous_element->angular_speed,
+                    current_element->time, current_element->angular_speed,
+                    trajectory_time);
+            const float interpolated_linear_speed = shared_tbots_math_linearInterpolation(
+                previous_element->time, previous_element->linear_speed,
+                current_element->time, current_element->linear_speed, trajectory_time);
 
             constant_period_trajectory->trajectory_elements[time_periods].position.x =
                 interpolated_x;
             constant_period_trajectory->trajectory_elements[time_periods].position.y =
                 interpolated_y;
-            constant_period_trajectory->speed_profile[time_periods] = interpolated_speed;
+            constant_period_trajectory->trajectory_elements[time_periods].orientation =
+                interpolated_theta;
+            constant_period_trajectory->trajectory_elements[time_periods].linear_speed =
+                interpolated_linear_speed;
+            constant_period_trajectory->trajectory_elements[time_periods].angular_speed =
+                interpolated_angular_speed;
             constant_period_trajectory->trajectory_elements[time_periods].time =
                 trajectory_time;
```

Comment:
Could we pull just a tad more out so we can write:
```
PositionTrajectoryElement_t* current_element =
    &variable_time_trajectory->trajectory_elements[i];
PositionTrajectoryElement_t* previous_element =
    &variable_time_trajectory->trajectory_elements[i - 1];

constant_period_trajectory->trajectory_elements[time_periods] = 
    app_trajectory_manager_interpolatePositionTrajectoryElements(
        current_element,
        previous_element,
        trajectory_time
        );
```
?

Reply:
Outdated.